### PR TITLE
Delete all other NCNs and save editor provided NCN to identifiers on document save

### PR DESCRIPTION
**Pull request by dragon-dxw**
## Jira card / Rollbar error (etc)
https://national-archives.atlassian.net/browse/FCL-496
image

### CODE REVIEW

File: judgments/views/judgment_edit.py [hunk 1]
```diff
@@ -50,6 +51,13 @@ def post(self, request, *args, **kwargs):
             new_citation = request.POST["neutral_citation"]
             api_client.set_judgment_citation(judgment_uri, new_citation)
 
+            # Set neutral citation in the identifiers
+            # Note that this currently does not preserve the UUID if the
+            # neutral citation is unchanged.
+            judgment.identifiers.delete_type(NeutralCitationNumber)
+            judgment.identifiers.add(NeutralCitationNumber(value=new_citation))
+            judgment.identifiers.save(judgment)
```
[the remark this answers is not in the document]
Comment (dragon-dxw) [reply judgments/views/judgment_edit.py]: Changes made, see also https://github.com/nationalarchives/ds-caselaw-custom-api-client/pull/801

File: judgments/tests/test_document_edit.py [hunk 2]
```diff
@@ -65,7 +66,7 @@ def test_edit_judgment(self, mock_judgment, api_client):
         )
         assert "NeutralCitationNumber" in str(judgment.identifiers.delete_type.call_args_list[0][0])
         assert "<Neutral Citation Number [4321] UKSC 123:" in str(judgment.identifiers.add.call_args_list[0][0])
-        judgment.identifiers.save.assert_called_with(judgment)
+        judgment.save_identifiers.assert_called_with()
```
Comment (jacksonj04) [inline judgments/tests/test_document_edit.py:69]: I think this can just be `assert_called_once()`.

**Review by jacksonj04 — APPROVED** [review]
One tiny tweak, then 🚀

File: judgments/tests/test_document_edit.py (hunk 2 above)
Comment (dragon-dxw) [reply judgments/tests/test_document_edit.py:69]: This is arguably stronger and more explicit; not only that it's called but with nothing.

**Pull request merged**